File: samples/java/src/main/java/com/amazon/aws/partners/saasfactory/controller/HomeController.java
/**
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package com.amazon.aws.partners.saasfactory.controller;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.GetMapping;

import java.util.Map;

@Controller
public class HomeController {

    private static final Logger LOGGER = LoggerFactory.getLogger(HomeController.class);

    @GetMapping({"/", "/index.html"})
    public String helloWorld(Map<String, Object> model) {
        String tenantId = System.getenv("TENANT_ID");
        if (tenantId == null || tenantId.isEmpty()) {
            tenantId = "Unknown";
        }
        model.put("tenantId", tenantId);
        return "index";
    }

}

File: samples/dotnet-framework/SaaSBoostHelloWorld/Controllers/HomeController.cs
/*
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
using log4net;
using System;
using System.Web.Mvc;

namespace SaaSBoostHelloWorld.Controllers
{
    public class HomeController : Controller
    {
        private static readonly ILog LOGGER = LogManager.GetLogger(typeof(HomeController));

        // GET: Home
        public ActionResult Index()
        {
            LOGGER.Info("HomeController::Index");
            string tenantId = Environment.GetEnvironmentVariable("TENANT_ID");
            if (String.IsNullOrEmpty(tenantId))  {
                tenantId = "Unknown";
            }
            LOGGER.Info($"Setting Tenant ID to {tenantId}");
            ViewData["TenantId"] = tenantId;
            return View();
        }
    }
}